src/lightning/fabric/loggers/csv_logs.py

```diff
@@ -244,11 +244,12 @@ def _record_new_keys(self) -> Set[str]:
         self.metrics_keys.extend(new_keys)
         return new_keys
 
-    def _rewrite_with_new_header(self, fieldnames: List[str]) -> None:
```

Comment:
Why are you rewriting this logic entirely? This is needed for sure, see the test we had.

Reply:
Hmm I did it because in the case that it was used, it was re-writing first and then opening the file again and appending the new metrics. However, we can just rewrite the file right? (this is what I do here)